src/med_bench/estimation/base.py

```diff
@@ -0,0 +1,496 @@
+
+        Parameters
+        ----------
+        t       array-like, shape (n_samples)
+                treatment value for each unit, binary
+
+        m       array-like, shape (n_samples)
+                mediator value for each unit, here m is necessary binary and uni-
+                dimensional
+
+        x       array-like, shape (n_samples, n_features_covariates)
+                covariates (potential confounders) values
+
+        y       array-like, shape (n_samples)
+                outcome value for each unit, continuous
+
+        nuisances
+        """
+        pass
+
+    def _resize(self, t, m, x, y):
+        """Resize data for the right shape
+
+        Parameters
+        ----------
+        t       array-like, shape (n_samples)
+                treatment value for each unit, binary
+
+        m       array-like, shape (n_samples)
+                mediator value for each unit, here m is necessary binary and uni-
+                dimensional
+
+        x       array-like, shape (n_samples, n_features_covariates)
+                covariates (potential confounders) values
+
+        y       array-like, shape (n_samples)
+                outcome value for each unit, continuous
+        """
+        if len(y) != len(y.ravel()):
+            raise ValueError("Multidimensional y is not supported")
+        if len(t) != len(t.ravel()):
+            raise ValueError("Multidimensional t is not supported")
+
+        n = len(y)
+        if len(x.shape) == 1:
+            x.reshape(n, 1)
+        if len(m.shape) == 1:
+            m = m.reshape(n, 1)
+
+        if n != len(x) or n != len(m) or n != len(t):
+            raise ValueError(
+                "Inputs don't have the same number of observations")
+
+        y = y.ravel()
+        t = t.ravel()
+
+        return t, m, x, y
+
+    def _input_reshape(self, t, m, x):
+        """Reshape data for the right shape"""
+        if len(t.shape) == 1:
+            t = t.reshape(-1, 1)
+        if len(m.shape) == 1:
+            m = m.reshape(-1, 1)
+        if len(x.shape) == 1:
+            x = x.reshape(-1, 1)
+
+        return t, m, x
+
+    def _fit_treatment_propensity_x_nuisance(self, t, x):
+        """Fits the nuisance parameter for the propensity P(T=1|X)"""
+        classifier = clone(self.classifier)
+        self._classifier_t_x = classifier.fit(x, t)
+
+        return self
+
+    def _fit_treatment_propensity_xm_nuisance(self, t, m, x):
+        """Fits the nuisance parameter for the propensity P(T=1|X, M)"""
+        xm = np.hstack((x, m))
+        self._classifier_t_xm = self.classifier.fit(xm, t)
+
+        return self
+
+    # TODO : Enable any sklearn object as classifier or regressor
+    def _fit_mediator_nuisance(self, t, m, x):
+        """Fits the nuisance parameter for the density f(M=m|T, X)"""
+        # estimate mediator densities
+        clf_param_grid = {}
+        classifier_m = GridSearchCV(self.classifier, clf_param_grid)
```

Comment:
I don't think we need this step. something like
```
classifier_m = clone(self.classifier)
```
?
true at many places, todo now or in the next PR?